nesta/packages/mag/query_mag_sparql.py

```diff
@@ -224,6 +259,61 @@ def update_field_of_study_ids_sparql(session, fos_ids):
     return fos_not_found
 
 
+def query_authors(ids=None, results_limit=None):
+    """Queries the MAG for authors and their affiliations.
+
+    Args:
+        ids: (:obj:`list` of `int`): author ids to query, all are returned if None
+        results_limit (int): limit the number of results returned (for testing)
+
+    Returns:
+        (dict): yields a single author at a time, with affiliation
+    """
+    query = '''
+    PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+    PREFIX foaf: <http://xmlns.com/foaf/0.1/>
+    PREFIX org: <http://www.w3.org/ns/org#>
+    PREFIX magp: <http://ma-graph.org/property/>
+    PREFIX magc: <http://ma-graph.org/class/>
+
+    SELECT ?author
+           ?authorName
+           ?affiliation
+           ?affiliationName
+           ?gridId
+    WHERE {{
+        ?author rdf:type magc:Author .
+        ?author foaf:name ?authorName .
+        ?author org:memberOf ?affiliation .
+        ?affiliation magp:grid ?gridId .
+        ?affiliation foaf:name ?affiliationName .
+        {}
+    }}
+    '''
+    concat_format = "<http://ma-graph.org/entity/{}>"
+
+    logging.debug(f"Querying MAG for authors: {ids}")
+    for count, row in enumerate(_batch_query_sparql(query,
+                                                    concat_format=concat_format,
+                                                    filter_on='author',
+                                                    ids=ids), start=1):
+        # rename and reformat fields out of entity urls
+        row['author_id'] = extract_entity_id(row.pop('author'))
```

Comment:
might be better to have this as an old-new field name mapping `dict`, which is an arg to `query_authors`.

Reply:
I've added the dict which made this look much neater, but I decided to keep it within the function for now as it's only relevant to the query written within that function. To be reviewed in future if there are more queries added.